test/integration/index.js

```diff
@@ -0,0 +1,205 @@
+      mockQuery
+        .bindQueryAndResponse(getVersionQuery, () =>
+          Promise.reject(
+            new Error(`relation "${versionsTableName}" does not exist`),
+          ),
+        )
+        .bindQueryAndResponse(BEGIN, () => Promise.resolve())
+        .bindQueryAndResponse(
+          createVersionsTableStatement,
+          () => Promise.resolve,
+        )
+        .bindQueryAndResponse(
+          new RegExp(`^INSERT INTO ${versionsTableName}.*`),
+          () => Promise.resolve(),
+        )
+        .bindQueryAndResponse(new RegExp('^sql-version-.*'), () =>
+          Promise.resolve(),
+        )
+        .bindQueryAndResponse(COMMIT, () => Promise.resolve());
+    });
+
+    describe('and dry run is set', () => {
+      beforeEach(() => {
+        set(context, DRY_RUN, true);
+      });
+
+      it('should dry run', () =>
+        expect(index(context)).to.eventually.be.fulfilled.then(() => {
+          expect(
+            mockLogger[get(context, ERROR_FN_LOGGER_CONFIG)].args,
+            'error-fn-calls',
+          ).to.deep.equal([]);
+
+          expect(
+            mockLogger[get(context, INFO_FN_LOGGER_CONFIG)].args,
+            'info-fn-calls',
+          ).to.deep.equal(
+            [[BEGIN], [createVersionsTableStatement]]
+              .concat(fromArray([A, B, C, D, E]))
+              .concat([[COMMIT]]),
+          );
+        }));
+    });
+
+    describe('and dry run is not set', () => {
+      it('should execute queries', () =>
+        expect(index(context)).to.eventually.be.fulfilled.then(() => {
+          expect(mockQuery.query.args).to.deep.equal(
+            [[getVersionQuery], [BEGIN], [createVersionsTableStatement]]
+              .concat(fromArray([A, B, C, D, E]))
+              .concat([[COMMIT]]),
+          );
+        }));
+
+      describe('and a database version exists', () => {
+        beforeEach(() => {
+          mockQuery = generateMockQuery(context)
+            .bindQueryAndResponse(getVersionQuery, () =>
+              Promise.resolve({ rows: [{ version: 0 }] }),
+            )
+            .bindQueryAndResponse(BEGIN, () => Promise.resolve())
+            .bindQueryAndResponse(
+              createVersionsTableStatement,
+              () => Promise.resolve,
+            )
+            .bindQueryAndResponse(
+              new RegExp(`^INSERT INTO ${versionsTableName}.*`),
+              () => Promise.resolve(),
+            )
+            .bindQueryAndResponse(new RegExp('^sql-version-.*'), () =>
+              Promise.resolve(),
+            );
+        });
+
+        describe('and no problems', () => {
+          beforeEach(() => {
+            mockQuery = mockQuery.bindQueryAndResponse(COMMIT, () =>
+              Promise.resolve(),
+            );
+          });
+
+          it('should execute expected queries', () =>
+            expect(index(context)).to.eventually.be.fulfilled.then(() => {
+              expect(mockQuery.query.args).to.deep.equal(
```

Comment:
where's the `args` come from?

Reply:
Here on line 20, I use spy (`sinon.spy`) to add instrumentation to the function like `.args` which gives me a list of all calls to that function.
https://github.com/0time/postgres-db-versioning/pull/1/files/208876c3a11cef60df1dfdabf8bf272ea7b469b6#diff-1bfce05ccae8d770f7484bca2d71faf6R20

As a bonus, here you can see that I reset mockQuery upon each test, so `.args` will never be polluted from one test to the next.
https://github.com/0time/postgres-db-versioning/pull/1/files/208876c3a11cef60df1dfdabf8bf272ea7b469b6#diff-cfce3706f03f2afd25608bfdae2bee4eR186